File: backend/api/src/main/scala/com/github/ai/split/api/response/PostGroupResponse.scala
package com.github.ai.split.api.response

import com.github.ai.split.api.GroupDto
import zio.json.{DeriveJsonEncoder, JsonEncoder}

case class PostGroupResponse(
  group: GroupDto
)

object PostGroupResponse {
  implicit val encoder: JsonEncoder[PostGroupResponse] = DeriveJsonEncoder.gen[PostGroupResponse]
}

File: backend/api/src/main/java/com/github/ai/split/api/response/PostGroupResponse.java
package com.github.ai.split.api.response;

import com.github.ai.split.api.GroupDto;
public class PostGroupResponse {

    public GroupDto group = new GroupDto();

    public PostGroupResponse() {}

    public PostGroupResponse(GroupDto group) {
        this.group = group != null ? group : new GroupDto();
    }
}